File: two-sum/two-sum.js
// https://leetcode.com/problems/two-sum/
// 44 ms 34.5 MB
var twoSum = function(nums, target) {
    var map = {};
    var len = nums.length;
    for(let i=0; i<len; i++){
        if(map[target-nums[i]] !== undefined){
            return [i, map[target-nums[i]]];
        }
        map[nums[i]]=i;
    }
    return [];
};

var ans1=twoSum([2, 7, 11, 15], 9);
console.log(ans1);

File: two-sum/two-sum.go
// https://leetcode.com/problems/two-sum/
// 4 ms	3.8 MB
package main
import "fmt"

func twoSum(nums []int, target int) []int {
    var m = make(map[int]int)
    for k, v := range nums{
        val, ok := m[target-v]
        if ok{
            return []int{val, k}
        }
        m[v] = k
    }
    return []int{}
}

func main(){
    num := []int{2, 7, 11, 15}
    target := 9
    fmt.Println(twoSum(num, target))
}
